gradio/blocks.py

```diff
@@ -304,39 +329,49 @@ def set_event_trigger(
         Context.root_block.dependencies.append(dependency)
         return dependency, len(Context.root_block.dependencies) - 1
 
-    def get_config(self):
-        return {
-            "visible": self.visible,
-            "elem_id": self.elem_id,
-            "elem_classes": self.elem_classes,
-            "root_url": self.root_url,
-        }
+    def get_config(self, with_globals=True):
+        config = {}
+        if with_globals:
+            config["root_url"] = self.root_url
+            config["name"] = self.get_block_name()
+        if hasattr(self.__class__, "__init__"):
+            signature = inspect.signature(self.__class__.__init__)
+            for k in signature.parameters:
+                if hasattr(self, k):
+                    v = getattr(self, k)
+                    config[k] = v
+
+        return config
 
     @staticmethod
-    @abstractmethod
     def update(**kwargs) -> dict:
```

Comment:
ahh dammit looks like we can't get rid of update from every component yet then, since we need to respect the order we set the arguments in the update method of Component :/

Reply:
Oh dang yea that's right the arg order in init vs update is sometimes different